In [68]:
import pandas as pd


In [69]:
df = pd.read_csv('finantialStatements(ms&tesla).csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6 entries, 0 to 5
Data columns (total 7 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   CompanyName       6 non-null      object
 1   Total_Revenue     6 non-null      object
 2   Years             6 non-null      int64 
 3   Net_Income        6 non-null      object
 4   Total_assets      6 non-null      object
 5   Total_Libilities  6 non-null      object
 6   Cash_flow_opera   6 non-null      object
dtypes: int64(1), object(6)
memory usage: 468.0+ bytes


In [70]:
df['Total_Revenue'] = df['Total_Revenue'].str.replace(',', '')  # Remove commas
df['Total_Revenue'] = df['Total_Revenue'].astype(float)         # Convert to float


In [71]:
df['Total_assets'] = df['Total_assets'].str.replace(',', '')  # Remove commas
df['Total_assets'] = df['Total_assets'].astype(float)         # Convert to float


In [72]:
df['Net_Income'] = df['Net_Income'].str.replace(',', '')  # Remove commas
df['Net_Income'] = df['Net_Income'].astype(float)         # Convert to float


In [73]:
df['Total_Libilities'] = df['Total_Libilities'].str.replace(',', '')  # Remove commas
df['Total_Libilities'] = df['Total_Libilities'].astype(float)         # Convert to float


In [74]:
df['Cash_flow_opera'] = df['Cash_flow_opera'].str.replace(',', '')  # Remove commas
df['Cash_flow_opera'] = df['Cash_flow_opera'].astype(float)         # Convert to float


In [75]:
df.head(7)

CompanyName  Total_Revenue  Years  Net_Income  Total_assets  \
0   Microsoft       245122.0   2024     88136.0      512163.0   
1   Microsoft       211915.0   2023     72316.0      411976.0   
2   Microsoft       198270.0   2022     72738.0      364840.0   
3       Tesla        97690.0   2024      7153.0      122070.0   
4       Tesla        96773.0   2023     14974.0      106618.0   
5       Tesla        81462.0   2022     12587.0       82338.0   

   Total_Libilities  Cash_flow_opera  
0          243686.0         118548.0  
1          205753.0          87582.0  
2          198298.0          89035.0  
3           48390.0          14923.0  
4           43009.0          13256.0  
5           36440.0          14724.0

In [76]:
df=df.sort_values(['CompanyName','Years'],ascending=[True,False])
df['RevenueGrowth(&)'] = df.groupby(df['CompanyName'])['Total_Revenue'].pct_change()*100
df

CompanyName  Total_Revenue  Years  Net_Income  Total_assets  \
0   Microsoft       245122.0   2024     88136.0      512163.0   
1   Microsoft       211915.0   2023     72316.0      411976.0   
2   Microsoft       198270.0   2022     72738.0      364840.0   
3       Tesla        97690.0   2024      7153.0      122070.0   
4       Tesla        96773.0   2023     14974.0      106618.0   
5       Tesla        81462.0   2022     12587.0       82338.0   

   Total_Libilities  Cash_flow_opera  RevenueGrowth(&)  
0          243686.0         118548.0               NaN  
1          205753.0          87582.0        -13.547132  
2          198298.0          89035.0         -6.438902  
3           48390.0          14923.0               NaN  
4           43009.0          13256.0         -0.938684  
5           36440.0          14724.0        -15.821562

In [77]:
df['NetIncomeGrowth(%)'] = df.groupby(['CompanyName'])['Net_Income'].pct_change() * 100
df

CompanyName  Total_Revenue  Years  Net_Income  Total_assets  \
0   Microsoft       245122.0   2024     88136.0      512163.0   
1   Microsoft       211915.0   2023     72316.0      411976.0   
2   Microsoft       198270.0   2022     72738.0      364840.0   
3       Tesla        97690.0   2024      7153.0      122070.0   
4       Tesla        96773.0   2023     14974.0      106618.0   
5       Tesla        81462.0   2022     12587.0       82338.0   

   Total_Libilities  Cash_flow_opera  RevenueGrowth(&)  NetIncomeGrowth(%)  
0          243686.0         118548.0               NaN                 NaN  
1          205753.0          87582.0        -13.547132          -17.949533  
2          198298.0          89035.0         -6.438902            0.583550  
3           48390.0          14923.0               NaN                 NaN  
4           43009.0          13256.0         -0.938684          109.338739  
5           36440.0          14724.0        -15.821562          -15.940964

## Finding Insights 
Financial Summary: Microsoft vs Tesla

### Microsoft:
Revenue declined consistently: -13.5% (2023), -6.4% (2022).

Net Income fell -17.9% in 2023, with slight recovery +0.6% in 2022.

Insight: Microsoft shows a downward trend in both revenue and profit, possibly due to rising costs or slowed growth.

### Tesla
Revenue dipped slightly: -0.9% (2023), and more in 2022: -15.8%.

Net Income surged +109.3% in 2023, then dropped -15.9% in 2022.

Insight: Tesla improved profitability in 2023 despite revenue dip, indicating stronger operational efficiency.